docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,199 @@ which groups these values were observed, as well as the difference and ratio
+
+The Solutionism Trap
+^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we assume that the best solution to a problem
+may involve technology, and fail to recognize other possible solutions outside of
+this realm. This trap also occurs as definitions of fairness are
+contested and changed and models become obsolete. See the "Construct
+Validity" section in Fairlearn's user guide.
+
+Example: consider the problem of internet connectivity in rural communities.
+An example of the solutionism trap is assuming that by using data science to
+study internet speed in a given region, policies and deals can be arranged to
+reduce the price of internet access. However, if there are additional socioeconomic
+problems within a community, for example lack of education, infrastructure, information
+technology and health services, then an algorithmic solution purely focused on internet
+speed will fail to meaningfully address the needs of the community.
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we do not consider the unintended consequences of introducing
+technology into an existing social system. Such consequences include change in
+behaviour, outcomes individual experience and a change in underlying social values
+and incentives of a given social system, for instance by increasing perceived value
+of quantifiable metrics over non-quantifiable ones.
+
+Example: consider the problem of banks deciding whether an individual should
+be approved for a loan. Prior to using machine learning algorithms
+to compute a "score", banks might rely on engaging in conversations with
+clients, recommending a plan based on their unique situation, and
+discussing with other team members to obtain feedback. By introducing an
+algorithm, it is possible that banks stop engaging in conversations
+with team members and clients, and assume the algorithm is good enough
+to blindly accept a recommendation. Furthermore, some might use
+information on people algorithm typically rejects to create a system
+that takes advantage of people who need funds but did not receive them.
+
+To avoid this pitfall, we must be aware that once a technology is incorporated
+into a social context, new groups may reinterpret it and potentially use this
+information for outside purposes which include nefarious ones. To
+help address this issue, we should adopt "what if" scenarios to envision how
+the social context might change after introducing a model, including how it may
+change the power dynamics of existing groups in that context.
+of the model.
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+Many tasks of a data scientist involve some form of formalization: from
+measuring real-world phenomena as data to translating business KPI's
+and constraints into metrics, loss functions, or parameters. We fall into the
+formalism trap when we fail to account for the full meaning of social
+concepts like fairness. This occurs because there is no purely mathematical way to resolve
```

Comment:
[if you have time]

This sentence suggests that we should somehow "resolve" conflicting definitions of fairness and that that's why we fall into the formalism trap. I think we can rephrase this paragraph a bit, e.g., something like:

"...concepts like fairness. 

Fairness is a complex construct that is contested: different people may have different ideas of what is fair in a particular scenario. While mathematical fairness metrics may capture some aspects of fairness, they fail to capture all relevant aspects. For example, group fairness metrics do not account for differences in individual experiences nor do they account for procedural justice.

In some scenarios, fairness metrics such as demographic parity and equalized odds cannot be satisfied at the same time. At a first glance, this may appear to be a mathematical problem. However, the conflict is actually grounded in different understandings of what fairness is. Consequently, there is no mathematical approach to solve the conflict. Instead we need to decide which metrics might are appropriate for the situation at hand, keeping in mind the limitations of a mathematical formalization. In some cases, there may be no suitable metric."

Reply:
I like this suggested revision @hildeweerts!

Reply:
This works. 
Incorporated changes.